### ISSUE TITLE
Kind is not able to start containers when userns-remap is enabled on the Docker daemon

### ISSUE THREAD

**Issue by vasrem** [kubernetes-sigs/kind#124]
Kind is not able to start containers when `userns-remap` is enabled on the Docker daemon. The error that occurs is the following:
```
time="13:37:34" level=error msg="docker: Error response from daemon: privileged mode is incompatible with user namespaces.  You must run the container in the host namespace when running privileged mode."
```

[Here](https://docs.docker.com/engine/security/userns-remap/#user-namespace-known-limitations) it  says that this is the correct behaviour.  

A solution could be that we add the flag `--userns=host` in the `docker run` commands that start the containers and contain the flag `--privileged`.

**Comment by BenTheElder** [comment]
we discussed this further via slack, and kind also needs a fix to the build to chown the tarballs at build time it looks like, will file a fix for that later today.

**Comment by BenTheElder** [comment]
the tarball change is in, but we've not pushed images built with it yet. locally built images should work `kind build node-image --type=apt && kind create cluster --image=kindest/node:latest`, but we still need to get #125.

**Comment by vasrem** [comment]
I updated the PR. It seems that `kind` doesn't work for docker daemon storage-driver different than `overlay2`. At least tested for `overlay` and `btrfs` and it doesn't work. Maybe this should be listed as a different Issue or in the documentation if you are not going to support these drivers. :)

**Comment by BenTheElder** [comment]
Yes, we should do that. btrfs is not going to work well I think, the others we might be able to fix / work around ... Originally it avoided this by using a tmpfs for the dind docker storage which avoided a lot of this but cost RAM :+)
So far `kind` is only tested with the default drivers on docker for mac, and docker-ce for debian + ubuntu, so I haven't made any claims there yet.
